Source/Scene/TileBoundingS2Cell.js

```diff
@@ -0,0 +1,695 @@
+    ]);
+  }
+
+  var center = s2Cell.getCenter();
+  centerCartographicScratch = ellipsoid.cartesianToCartographic(
+    center,
+    centerCartographicScratch
+  );
+  centerCartographicScratch.height = (maximumHeight + minimumHeight) / 2;
+  this.center = ellipsoid.cartographicToCartesian(
+    centerCartographicScratch,
+    center
+  );
+
+  this._boundingSphere = BoundingSphere.fromPoints(vertices);
+}
+
+var centerGeodeticNormalScratch = new Cartesian3();
+var topCartographicScratch = new Cartographic();
+var topScratch = new Cartesian3();
+var vertexCartographicScratch = new Cartographic();
+var vertexScratch = new Cartesian3();
+var vertexGeodeticNormalScratch = new Cartesian3();
+var sideNormalScratch = new Cartesian3();
+var sideScratch = new Cartesian3();
+var topPlaneScratch = new Plane(Cartesian3.UNIT_X, 0.0);
+var bottomPlaneScratch = new Plane(Cartesian3.UNIT_X, 0.0);
+/**
+ * Computes bounding planes of the kDOP.
+ * @private
+ */
+function computeBoundingPlanes(
+  s2Cell,
+  minimumHeight,
+  maximumHeight,
+  ellipsoid
+) {
+  var planes = new Array(6);
+  var centerPoint = s2Cell.getCenter();
+
+  // Compute top plane.
+  // - Get geodetic surface normal at the center of the S2 cell.
+  // - Get center point at maximum height of bounding volume.
+  // - Create top plane from surface normal and top point.
+  var centerSurfaceNormal = ellipsoid.geodeticSurfaceNormal(
+    centerPoint,
+    centerGeodeticNormalScratch
+  );
+  var topCartographic = ellipsoid.cartesianToCartographic(
+    centerPoint,
+    topCartographicScratch
+  );
+  topCartographic.height = maximumHeight;
+  var top = ellipsoid.cartographicToCartesian(topCartographic, topScratch);
+  var topPlane = Plane.fromPointNormal(
+    top,
+    centerSurfaceNormal,
+    topPlaneScratch
+  );
+  planes[0] = topPlane;
+
+  // Compute bottom plane.
+  // - Iterate through bottom vertices
+  //   - Get distance from vertex to top plane
+  // - Find longest distance from vertex to top plane
+  // - Translate top plane by the distance
+  var maxDistance = 0;
+  var i;
+  var vertex, vertexCartographic;
+  for (i = 0; i < 4; i++) {
+    vertex = s2Cell.getVertex(i);
+    vertexCartographic = ellipsoid.cartesianToCartographic(
+      vertex,
+      vertexCartographicScratch
+    );
+    vertexCartographic.height = minimumHeight;
+    var distance = Plane.getPointDistance(
+      topPlane,
+      ellipsoid.cartographicToCartesian(vertexCartographic, vertexScratch)
+    );
+    if (distance < maxDistance) {
+      maxDistance = distance;
+    }
+  }
+  var bottomPlane = Plane.clone(topPlane, bottomPlaneScratch);
+  // Negate the normal of the bottom plane since we want all normals to point "outwards".
+  bottomPlane.normal = Cartesian3.negate(
+    bottomPlane.normal,
+    bottomPlane.normal
+  );
+  bottomPlane.distance = bottomPlane.distance * -1 + maxDistance;
+  planes[1] = bottomPlane;
+
+  // Compute side planes.
+  // - Iterate through vertices
```

Comment:
This is done in CCW order, correct? maybe make a note of that since it seems relevant in `distanceToCamera`